### PR TITLE
Implement vm.Pause() directly

### PR DESCRIPTION
**Pull request by Crazykev**
/cc @laijs As you suggested in #437, PTAL.

Signed-off-by: Crazykev <crazykev@zju.edu.cn>

### CODE REVIEW

File: hypervisor/vm.go [hunk 1]
```diff
@@ -635,30 +640,20 @@ func (vm *Vm) Pause(pause bool) error {
...
-	}, StateRunning)
-
-	if oldPauseState == pauseState {
-		return nil
+		err = ctx.DCtx.Pause(ctx, pause)
 	}
 	if err != nil {
-		pauseState = oldPauseState // recover the state
+		glog.Errorf("%s sandbox failed, recover the state", command)
+		ctx.PauseState = oldPauseState // recover the state
```
Comment (laijs) [inline hypervisor/vm.go]: the above line and oldPauseState should be removed, ctx.PauseState has not been changed.

but you forgot to set ctx.PauseState after success.

Comment (Crazykev) [inline hypervisor/vm.go]: Indeed, the original design lead me to expect ctx.DCtx.Pause() will change the state. Fixed and add a comment there.

**Comment by laijs** [comment]
could you also fix the commit title

File: hypervisor/vm.go [hunk 2]
```diff
@@ -627,39 +627,35 @@ func (vm *Vm) Stats() *types.PodStats {
...
+		return err
 	}
-	vm.GenericOperation(command+" result", func(ctx *VmContext, result chan<- error) {
-		ctx.PauseState = pauseState
-		result <- nil
-	}, StateRunning)
-	return err
+
+	glog.V(3).Infof("sandbox state turn to %s now", command)
+	ctx.PauseState = pauseState // change the state.
```
Comment (laijs) [inline hypervisor/vm.go]: the above code should be moved up into the branch 'if ctx.PauseState != pauseState'

**Comment by Crazykev** [comment]
@laijs Fixed.

**Comment by laijs** [comment]
could you also use the pauseLock on Save()?

**Comment by Crazykev** [comment]
@laijs Sure, will do.

**Pull request merged**